File: Player.cpp
// player.cpp

#define _CRT_SECURE_NO_WARNINGS
#include "Player.h"

Player::Player(const std::string& name) : name(name), score(0) {}

std::string Player::getName() const { return name; }
int Player::getScore() const { return score; }
void Player::setName(const std::string& name) { this->name = name; }
void Player::incrementScore() { score++; }
void Player::resetScore() { score = 0; }
void Player::printInfo() const {
    std::cout << "Игрок: " << name << ", Счёт: " << score << std::endl;
}

void Player::specialAbility() const {
    std::cout << "У игрока нет специальных способностей." << std::endl;
}

std::ostream& operator<<(std::ostream& os, const Player& player) {
    os << "Игрок: " << player.name << ", Cчёт: " << player.score;
    return os;
}

File: Player.java
// player.java

/**
 * Класс, представляющий игрока в игре.
 */
public class Player {
    protected String name;  // Имя игрока
    protected int score;    // Очки игрока
    private static int playerCount = 0;  // Общее количество игроков

    public Player() {
        this.name = "";
        this.score = 0;
        playerCount++;
    }

    public Player(String name) {
        this.name = name;
        this.score = 0;
        playerCount++;
    }

    public String getName() {
        return name;
    }

    public int getScore() {
        return score;
    }

    public void incrementScore() {
        this.score++;  // Увеличение очков на 1
    }

    public void resetScore() {
        this.score = 0;  // Сброс очков
    }

    public void setName(String name) {
        this.name = name;  // Установка имени
    }

    public static int getPlayerCount() {
        return playerCount;  // Возвращает количество игроков
    }

    @Override
    public String toString() {
        return "Player{name='" + name + "', score=" + score + "}";
    }
}
